Add README and sample documents

vladimir-v-diaz opened the pull request:
* **What does this PR do?**

It adds sample documents for things like reporting vulnerabilities, project governance, configuration files, etc.  The README summarizes the documents available on this repo.

* **Are there points in the code the reviewer needs to double check?**
N/A.

* **Who do you think should review this PR?**
The software developers and any interested research students in our lab.
@awwad @lukpueh

* **Why was this PR needed?**
All of our projects should strive to meet the requirements for the Core Infrastructure Initiative Best  Practices Badge.  These sample documents can be used to satisfy some of these requirements. 

* **What are the relevant issue numbers fixed (add one line for each)?**

Fixes #1.

vladimir-v-diaz committed:
Add initial templates

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz committed:
Add README

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz committed:
List issue and pull request GitHub templates

vladimir-v-diaz committed:
Add summary for vulnerability reports and maintainers

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz committed:
Add text for GitHub issue and pull request templates

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz committed:
Add remaining text for the README

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz commented:
This pull request is now ready for review!  A README was added, which includes a summary of each template.

JustinCappos commented on ACKNOWLEDGEMENTS.md:
```diff
@@ -0,0 +1,8 @@
+Systems Lab at NYU.
+
+This material is based upon work supported by the National Science Foundation
+under Grant Nos. CNS-1345049 and CNS-0959138. Any opinions, findings, and
```
Should this still be here as an example or should we remove these so that others fill them in?

JustinCappos replied:
This is meant to be a more general comment, whether we should have this more as a template or an example.  I can see pros and cons to both.

vladimir-v-diaz replied:
I guess these are more examples than template files.  Perhaps we can edit the wording in the README to make this clear.  What do you think?

Some of the other "templates" suffer from the same shortcoming.  For example, the travis configuration file would need to be edited, because not all projects use the same branch names, test suite, etc.

I hope that other projects can use these examples as a starting point.

vladimir-v-diaz replied:
The README was revised to use "sample documents" instead of "templates".  Please have a look at the new README and let me know if it works better.  If so, I can go ahead and rename the repository, as well.

JustinCappos replied:
I think perhaps also making some edits in the files to show where to change things is wise.  (If you disagree, let's discuss.)

vladimir-v-diaz committed:
Replace before_script entries in .travis.yml

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz committed:
Revise README.  templates -> sample documents

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz committed:
Revise text on maintainers

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz committed:
Miscellaneous edits to text

vladimir-v-diaz committed:
Revise and reorganize parts of the README

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz committed:
Add notice to many of the sample documents

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz committed:
Add notice to remaining sample documents

Signed-off-by: Vladimir Diaz <vladimir.v.diaz@gmail.com>

vladimir-v-diaz commented:
@JustinCappos 
I renamed the repository and revised the repo description, added a notice to all sample documents, and revised the README.  Let me know if the pull request is ready to be merged.

JustinCappos commented:
Okay, I think this is good enough for now.  We can improve it later if people find it confusing...

vladimir-v-diaz commented:
Note: I added text to the README encouraging the reader to explore the TUF repo, which can demonstrate how some of these sample documents are used in practice: https://github.com/secure-systems-lab/sample-documents/commit/ca6c3d8d00fe28825466c635656cef8988a4032d
